cpp/src/arrow/stl_iterator.h

```diff
@@ -59,11 +62,12 @@ class ArrayIterator {
 
   // Value access
   value_type operator*() const {
-    return array_->IsNull(index_) ? value_type{} : array_->GetView(index_);
+    return !array_ || array_->IsNull(index_) ? value_type{} : array_->GetView(index_);
```

Comment:
Is there any circumstance where `array_` is null and you're still calling this operator? This doesn't seem right.

Reply:
When a `ChunkedArray` is empty, I initialize an `ArrayIterator` with the empty constructor which initialize the `array_` with a `NULLPTR`, and this extra condition is more feasible/short than adding more conditions on ChunkedArrayIterator for validate multiple cases.

Reply:
I don't think dereferencing such an iterator should be a valid operation, though, so the check should not be needed.